config/spl-build.m4

```diff
@@ -1654,3 +1655,29 @@ AC_DEFUN([SPL_AC_KERNEL_READ], [
 	])
 	EXTRA_KCFLAGS="$tmp_flags"
 ])
+
+dnl #
+dnl # 4.15 API change
+dnl # https://lkml.org/lkml/2017/11/25/90
+dnl # Check if timer_list.func get passed a timer_list or an unsigned long
+dnl # (older kernels).
```

Comment:
In addition to checking for the callback prototype can you also verify `from_timer()` and `timer_setup()` in this `configure` check.  Let's be strict about only using this new interface when everything is exactly as expected.